frontend/src/component/common/ApiTokenTable/useApiTokenTable.tsx

```diff
@@ -0,0 +1,145 @@
+                Cell: () => <IconCell icon={<Key color="disabled" />} />,
+                disableSortBy: true,
+                disableGlobalFilter: true,
+            },
+            {
+                Header: 'Username',
+                accessor: 'username',
+                Cell: HighlightCell,
+            },
+            {
+                Header: 'Type',
+                accessor: 'type',
+                Cell: ({
+                    value,
+                }: {
+                    value: 'client' | 'admin' | 'frontend';
+                }) => (
+                    <HighlightCell
+                        value={tokenDescriptions[value.toLowerCase()].label}
+                        subtitle={tokenDescriptions[value.toLowerCase()].title}
+                    />
+                ),
+                minWidth: 280,
+            },
+            {
+                Header: 'Project',
+                accessor: 'project',
+                Cell: (props: any) => (
+                    <ProjectsList
+                        project={props.row.original.project}
+                        projects={props.row.original.projects}
+                    />
+                ),
+                minWidth: 120,
+            },
+            {
+                Header: 'Environment',
+                accessor: 'environment',
+                Cell: HighlightCell,
+                minWidth: 120,
+            },
+            {
+                Header: 'Created',
+                accessor: 'createdAt',
+                Cell: DateCell,
+                minWidth: 150,
+                disableGlobalFilter: true,
+            },
+            {
+                Header: 'Last seen',
+                accessor: 'seenAt',
+                Cell: TimeAgoCell,
+                minWidth: 150,
+                disableGlobalFilter: true,
+            },
+            {
+                Header: 'Actions',
+                id: 'Actions',
+                align: 'center',
+                width: '1%',
+                disableSortBy: true,
+                disableGlobalFilter: true,
+                Cell: getActionCell,
+            },
+        ];
+    }, []);
+
+    const {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        prepareRow,
+        state,
+        setGlobalFilter,
+        setHiddenColumns,
+    } = useTable(
+        {
+            columns: COLUMNS as any,
+            data: tokens as any,
+            initialState,
+            sortTypes,
+            autoResetHiddenColumns: false,
+            disableSortRemove: true,
+        },
+        useGlobalFilter,
+        useSortBy
+    );
+
+    return {
+        getTableProps,
+        getTableBodyProps,
+        headerGroups,
+        rows,
+        prepareRow,
+        state,
+        setGlobalFilter,
+        setHiddenColumns,
+        columns: COLUMNS,
+    };
+};
+
+const tokenDescriptions: { [index: string]: { label: string; title: string } } =
```

Comment:
can we narrow down index: string to index: 'client' | 'frontend' | 'admin'?

Reply:
Good idea